p2p/host/basic/basic_host.go

```diff
@@ -276,7 +315,28 @@ func New(net network.Network, opts ...interface{}) *BasicHost {
 // Start starts background tasks in the host
 func (h *BasicHost) Start() {
 	h.refCount.Add(1)
-	go h.background()
+	go h.listenAddrChangeLoop()
+
+	h.refCount.Add(1)
+	go h.localIpAddrChangeLoop()
+}
+
+func (h *BasicHost) localIpAddrChangeLoop() {
```

Comment:
how painful would it be to combine this into the select loop for the addrChangeTicker and save the extra goroutine?

Reply:
@willscott The code will look a bit messy. This keeps things simple to read which I like.

Reply:
I have to disagree. Chasing down the channel to figure out what it triggers is harder to read.

Can we just re-use the same ticker interval? Once every five seconds shouldn't be too inefficient. We can just call `updateLocalIpAddr` right before https://github.com/libp2p/go-libp2p/pull/936/files#diff-439e1c54aa40f8126fab7c895858fae3R492.